### ISSUE TITLE
Customization of message-to-alert mapping

### ISSUE THREAD

**Issue by greyhare** [zostera/django-bootstrap4#241]
In `bootstrap4/templatetags/bootstrap4.py`, there's a function `bootstrap_message_classes` which figures out what classes to assign to a Bootstrap Alert generated from a Django message. Right now, it makes the obvious mappings, except for two:

* `message_constants.DEBUG` is mapped to `alert-warning`, and
* messages without a proper level set get `alert-danger`.

Personally, I'd like `DEBUG` mapped to `alert-secondary` (gray background) and the default to be `alert-primary` (darker blue background) because I don't want things that aren't warnings or errors to look like them. Others might want to use `alert-light` or `alert-dark`. But there's no obvious way to change this. Monkeypatching the function might not work because `@register.filter` would still register the old one. Maybe I could alter `MESSAGE_LEVEL_CLASSES`, but is that the right way?

**Comment by greyhare** [comment]
I've gotten this to work in my `settings.py`, but I have no idea how fragile it is:

```python
from bootstrap4.templatetags import bootstrap4
from django.contrib.messages import constants
bootstrap4.MESSAGE_LEVEL_CLASSES[constants.DEBUG] = 'alert alert-secondary'
bootstrap4.MESSAGE_LEVEL_CLASSES[99] = 'alert alert-primary'
```

The `99` is the value I use to test the default case. Needless to say, this doesn't really change the default styling, and one would need to find all the messages that didn't use `constants` and set values for them in the dict.

This really needs to be something we can set in `settings.BOOTSTRAP4`.

**Comment by dyve** [comment]
Makes sense to be able to set this. I'll also check the django-bootstrap5 version of this issue and see what I can come up with.

**Comment by dyve** [comment]
Reasonable request, but it needs a new `BOOTSTRAP4` setting, and this package is in maintenance mode: bug fixes and security updates only.

Two things work today. Your own recipe, mutating the dict at import time, is supported in practice because `bootstrap_message_classes` reads `MESSAGE_LEVEL_CLASSES` on every call, so it is less fragile than you feared:

```python
from django.contrib.messages import constants
from bootstrap4.templatetags import bootstrap4
bootstrap4.MESSAGE_LEVEL_CLASSES[constants.DEBUG] = "alert alert-secondary"
```

Cleaner is to override `bootstrap4/messages.html` in your own templates directory. You then control the classes and the markup without depending on module internals, and the default case is yours too.

django-bootstrap5 is the place for the settings-based version.

Closing as out of scope. Sorry it took this long to give you a straight answer.

**Comment by greyhare** [comment]
I moved to Bootstrap 5 in the interim. ^_^
